src/plugins/dashboard/public/application/lib/save_dashboard.ts

```diff
@@ -31,35 +31,30 @@
 import { TimefilterContract } from 'src/plugins/data/public';
 import { SavedObjectSaveOpts } from 'src/plugins/saved_objects/public';
 import { updateSavedDashboard } from './update_saved_dashboard';
-import { DashboardStateManager } from '../dashboard_state_manager';
+
+import { DashboardAppStateContainer } from '../../types';
+import { Dashboard } from '../../dashboard';
 
 /**
  * Saves the dashboard.
- * @param toJson A custom toJson function. Used because the previous code used
- * the angularized toJson version, and it was unclear whether there was a reason not to use
- * JSON.stringify
  * @returns A promise that if resolved, will contain the id of the newly saved
  * dashboard.
  */
 export function saveDashboard(
-  toJson: (obj: any) => string,
   timeFilter: TimefilterContract,
-  dashboardStateManager: DashboardStateManager,
-  saveOptions: SavedObjectSaveOpts
+  stateContainer: DashboardAppStateContainer,
+  savedDashboard: any,
+  saveOptions: SavedObjectSaveOpts,
+  dashboard: Dashboard
 ): Promise<string> {
-  const savedDashboard = dashboardStateManager.savedDashboard;
-  const appState = dashboardStateManager.appState;
+  const appState = stateContainer.getState();
 
-  updateSavedDashboard(savedDashboard, appState, timeFilter, toJson);
+  updateSavedDashboard(savedDashboard, appState, timeFilter, dashboard);
```

Comment:
dashboard is also passed here because we also need to not only need to update the values in saved dashboard instance but also in the dashboard class. So dashboard class can have the latest values after the action of save. 

One todo is to probably write another function called updateDashboard() so the logics for updating the two can be separated.